### PR TITLE
Add gum_arm64_writer_put_add_reg_reg_reg function

### PR DESCRIPTION
**Pull request by Tyilo**
This is needed for the arm64 fix in frida-core.

### CODE REVIEW

**Comment by Tyilo** [comment]
I haven't actually tested this, but I think it should work.

File: gum/arch-arm64/gumarm64writer.c [hunk 1]
```diff
@@ -707,6 +707,29 @@ gum_arm64_writer_put_add_reg_reg_imm (GumArm64Writer * self,
...
+
+  gum_arm64_writer_describe_reg (self, dst_reg, &rd);
+  gum_arm64_writer_describe_reg (self, left_reg, &rl);
+  gum_arm64_writer_describe_reg (self, right_reg, &rr);
+
+  g_assert_cmpuint (rd.width, ==, rl.width);
+  g_assert_cmpuint (rd.width, ==, rr.width);
+
+  guint32 flags = 0;
+  if(rd.width == 64) {
```
Comment (oleavr) [inline gum/arch-arm64/gumarm64writer.c]: Two nitpicks: should be space between `if` and `(`, and curly braces should align, though in this case they should be omitted to make the code more readable.

File: gum/arch-arm64/gumarm64writer.c [hunk 2]
```diff
@@ -707,6 +707,29 @@ gum_arm64_writer_put_add_reg_reg_imm (GumArm64Writer * self,
...
+    flags |= 0x8000000;
+  }
+
+  gum_arm64_writer_put_instruction (self, rd.sf | 0xb000000 | flags | rd.index | (rl.index << 5) | (rr.index << 16));
```
Comment (oleavr) [inline gum/arch-arm64/gumarm64writer.c]: Would be great to break the line at 80 cols (see style elsewhere in this file). Note that `frida-core` doesn't have an 80 col limit, as the plan is to move `frida-gum` to the same conventions later; but for now let's try to keep things consistent within each module.

File: gum/arch-arm64/gumarm64writer.c [hunk 3]
```diff
@@ -707,6 +707,29 @@ gum_arm64_writer_put_add_reg_reg_imm (GumArm64Writer * self,
...
+  GumArm64RegInfo rd, rl, rr;
+
+  gum_arm64_writer_describe_reg (self, dst_reg, &rd);
+  gum_arm64_writer_describe_reg (self, left_reg, &rl);
+  gum_arm64_writer_describe_reg (self, right_reg, &rr);
+
+  g_assert_cmpuint (rd.width, ==, rl.width);
+  g_assert_cmpuint (rd.width, ==, rr.width);
+
+  guint32 flags = 0;
```
Comment (oleavr) [inline gum/arch-arm64/gumarm64writer.c]: Would be good to move the declaration just after the `rd, rl, rr`, as declaration-after-statement is avoided through-out frida-gum due to Microsoft's terrible compiler. (I tend to declare variables in the same order as they're used.)

**Comment by oleavr** [comment]
Yay, awesome!

**Pull request merged**